### PR TITLE
Added Jukebox

### PR DESCRIPTION
**Pull request by metalgearsloth**
Resurrected from inverted's original PR which died due to requiring audio rework which is now done.

Draft so I can fix a bunch of stuff and check it works. Requires https://github.com/space-wizards/RobustToolbox/pull/5024

Lag compensated + audio is fully networked.

https://github.com/space-wizards/space-station-14/assets/31366439/f2b7609a-7eac-41e2-8a3b-5212d111413d

:cl: iNV3RT3D & metalgearsloth
- add: Added a jukebox.

### CODE REVIEW

**Comment by ps3moira** [comment]
[https://www.youtube.com/watch?v=u8Ou884gD_w](url) bar's gonna be groovy

File: Content.Client/Audio/Jukebox/JukeboxBoundUserInterface.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    protected override void UpdateState(BoundUserInterfaceState state)
+    {
+        base.UpdateState(state);
+
+        if (state is not JukeboxBoundInterfaceState bState || _menu == null)
+            return;
+
+        // TODO: Need a way to sub to compstates as otherwise this can fail in some situations.
```
Comment (DrSmugleaf) [inline Content.Client/Audio/Jukebox/JukeboxBoundUserInterface.cs:61]: Can't you subscribe to the after handle auto comp state event for this?

File: Content.Shared/Audio/Jukebox/JukeboxUi.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+public enum JukeboxUiKey : byte
+{
+    Key,
+}
+
+[Serializable, NetSerializable]
+public sealed class JukeboxBoundInterfaceState : BoundUserInterfaceState
+{
+    public NetEntity? Audio;
+    public ProtoId<JukeboxPrototype>? SelectedSong;
```
Comment (DrSmugleaf) [inline Content.Shared/Audio/Jukebox/JukeboxUi.cs:17]: These are already on the jukebox component, can the client UI use the values from that instead?

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

**Comment by K-Dynamic** [comment]
Now we just need a dj table to create ear-destroying nightcore mixes with added cowbells

**Pull request merged**